packages/editor/src/editor/canvas/ComponentBlock.tsx

```diff
@@ -42,12 +42,29 @@ const Draggable = ({ config, data }: DraggableProps) => {
   const { setUi } = useAppContext();
   const { data: formData, setData } = useData();
   const readonly = useReadonly();
-  const { isDragging, attributes, listeners, setNodeRef } = useDraggable({ disabled: readonly, id: data.cid, data: dragData(data) });
+  const isDataTableEditableButtons =
+    data.type === 'Button' && ((data.config as ButtonType).type === 'EDIT' || (data.config as ButtonType).type === 'DELETE');
+  const { isDragging, attributes, listeners, setNodeRef } = useDraggable({
+    disabled: readonly || isDataTableEditableButtons,
+    id: data.cid,
+    data: dragData(data)
+  });
   const { selectedElement, setSelectedElement } = useAppContext();
   const isSelected = selectedElement === data.cid;
   const elementConfig = { ...config.defaultProps, ...data.config };
   const deleteElement = () => {
     setData(oldData => modifyData(oldData, { type: 'remove', data: { id: data.cid } }).newData);
+    if (data.type === 'DataTable' && (data.config as DataTable).editDialogId.length > 0) {
+      setData(
+        oldData =>
+          modifyData(oldData, {
+            type: 'remove',
+            data: {
+              id: (data.config as DataTable).editDialogId
+            }
+          }).newData
+      );
+    }
```

Comment:
Hmm understandable why this code is needed. However I do not really like that this lives here. It should live on the datatable somewhere.
E.g. a config on the component config which can be defined by the components what should all happen if it gets deleted.